src/GoogleAuthService/index.js

```diff
@@ -74,13 +74,13 @@ export default class GoogleAuthService {
    * @return {Promise<string>} authorization code
    */
   grantOfflineAccess() {
-    return this.getAuthInstance()
-      .grantOfflineAccess()
-      .then(({ code }) => {
+    return this.getAuthInstance().then((authInstance) => {
+      return authInstance.grantOfflineAccess().then(({ code }) => {
```

Comment:
@cedpoilly, I'm not as familiar with this functionality (and broke it in v1.0 because I failed to test it 😬), but I'm wondering how this used to work within the login flow?

It seems `grantOfflineAccess` opens _another_ popup which is getting blocked for me when being invoked immediately after `signIn` 🤔

Reply:
I did a bit of research (because I am also not familiar with this) and can confirm your observation: `grantOfflineAccess` intentionally opens another popup because the user should confirm that the app can renew the access token infinitely.
Because the popup is triggered asynchronously (after `signIn`), browsers block it. `grantOfflineAccess` should be called synchronously to a user interaction.
See https://github.com/google/google-api-javascript-client/issues/333#issuecomment-336950810 and https://github.com/google/google-api-javascript-client/issues/664#issuecomment-672810512

[This comment](https://github.com/google/google-api-javascript-client/issues/306#issuecomment-651566890) suggests that `grantOfflineAccess` can be used instead of `signIn`.

Reply:
Interesting. Thanks for looking into this, @fmeyertoens!

I imagine we could do something like https://github.com/google/google-api-javascript-client/issues/306#issuecomment-651566890 when `login`'s `grantOfflineAccess=true` – and maybe make `GoogleAuthService.grantOfflineAccess` private for clarity?

Though, it would probably be ideal to collect some more feedback from others who were using this functionality before making a change.

Reply:
Two thoughts on that.
First, the comment mentions that this is a workaround and I don't understand all implications of that.
Second, the [`GoogleAuth.grantOfflineAccess`](https://developers.google.com/identity/sign-in/web/reference#googleauthgrantofflineaccessoptions) function can take an options object which we don't expose at the moment. Actually the only thing we do is storing the code in the session storage (which I don't know is any useful since the received auth code has to be exchanged for a token anyway). We might take this config object and pass it trough or devs could just get the `authInstance` and call `grantOfflineAccess` on it themselves.

Reply:
I sort of forgot about about this but realize it's still broken.

Do we have any additional thoughts on how we should handle this?

Reply:
99de043e6092821d9ed98df9c0e6781582d0e25e just removes the (undocumented) `grantOfflineAccess` login option which doesn't seem to work per the above, which I think is the simplest option for now. Users will presumably expose a second button post-login to support this flow, invoking `grantOfflineAccess` directly.

This is technically a breaking change, but I think a patch is fine given that it wasn't working before.